diff --git a/js/entity_browser.modal.js b/js/entity_browser.modal.js
index 9418e07..f7d601c 100644
--- a/js/entity_browser.modal.js
+++ b/js/entity_browser.modal.js
@@ -119,7 +119,9 @@
         }
 
         // Reposition dialog.
-        dialog.option('position', dialog.options.position);
+        setTimeout(function() {
+          dialog.option('position', dialog.options.position);
+        }, 500);
       }
     });
   };
