diff --git a/tmgmt_contentapi/src/Swagger/Client/Configuration.php b/tmgmt_contentapi/src/Swagger/Client/Configuration.php
index 2838778..6b9f050 100644
--- a/tmgmt_contentapi/src/Swagger/Client/Configuration.php
+++ b/tmgmt_contentapi/src/Swagger/Client/Configuration.php
@@ -87,7 +87,7 @@ class Configuration
      *
      * @var string
      */
-    protected $host = 'https://contentapi.lionbridge.com/v2';
+    protected $host = 'https://content-api.staging.lionbridge.com/v2';

     /**
      * The host_ctt
