diff --git a/sources/content/tmgmt_content.module b/sources/content/tmgmt_content.module
index 23d526c..105695d 100644
--- a/sources/content/tmgmt_content.module
+++ b/sources/content/tmgmt_content.module
@@ -201,7 +201,6 @@ function tmgmt_content_form_tmgmt_settings_form_alter(array &$form, FormStateInt
       '#type' => 'item',
       '#title' => t('Always embedded'),
       '#description' => t('These references are always embedded in the translatable data.'),
-      '#description_display' => 'after',
       'list' => array(
         '#theme' => 'item_list',
         '#items' => $always_embedded,
