diff --git a/src/Plugin/Block/SimpleInstagramBlock.php b/src/Plugin/Block/SimpleInstagramBlock.php
index c960ec8..85f60ae 100644
--- a/src/Plugin/Block/SimpleInstagramBlock.php
+++ b/src/Plugin/Block/SimpleInstagramBlock.php
@@ -35,21 +35,20 @@ class SimpleInstagramBlock extends BlockBase implements BlockPluginInterface {
       '#type' => 'checkbox',
       '#title' => $this->t('Display profile?'),
       '#description' => $this->t('Do you wish to display the Instagram profile on this Instagram Feed?'),
-      '#default_value' => isset($config['simple_instagram_display_profile']) ? $config['simple_instagram_display_profile'] : 'true',
+      '#default_value' => isset($config['simple_instagram_display_profile']) ? $config['simple_instagram_display_profile'] : 'false',
     ];
 
     $form['simple_instagram_display_biography'] = [
       '#type' => 'checkbox',
       '#title' => $this->t('Display bio?'),
       '#description' => $this->t('Do you wish to display the Instagram Bio on this Instagram Feed?'),
-      '#default_value' => isset($config['simple_instagram_display_biography']) ? $config['simple_instagram_display_biography'] : 'true',
+      '#default_value' => isset($config['simple_instagram_display_biography']) ? $config['simple_instagram_display_biography'] : 'false',
     ];
 
     $form['simple_instagram_items'] = [
-      '#type' => 'textfield',
-      '#size' => 3,
-      '#maxlength' => 3,
-      '#title' => $this->t('Number of images'),
+      '#type' => 'select',
+      '#options' => ['3' => '3', '6' => '6', '9' => '9', '12' => '12'],
+      '#title' => $this->t('Number of images?'),
       '#description' => $this->t('How many images do you wish to feature on this Instagram Feed?'),
       '#default_value' => isset($config['simple_instagram_items']) ? $config['simple_instagram_items'] : '12',
       '#required' => TRUE,
@@ -61,7 +60,7 @@ class SimpleInstagramBlock extends BlockBase implements BlockPluginInterface {
       '#options' => [$simple_items_range],
       '#title' => $this->t('Number of images per row?'),
       '#description' => $this->t('How many images do you wish to feature on each row of this Instagram Feed? You can produce a single row if you set the numnber of images to equal the number of images per row.'),
-      '#default_value' => isset($config['simple_instagram_items_per_row']) ? $config['simple_instagram_items_per_row'] : '5',
+      '#default_value' => isset($config['simple_instagram_items_per_row']) ? $config['simple_instagram_items_per_row'] : '2',
     ];
 
     $form['simple_instagram_styling'] = [
@@ -69,7 +68,7 @@ class SimpleInstagramBlock extends BlockBase implements BlockPluginInterface {
       '#options' => ['true' => 'True', 'false' => 'False'],
       '#title' => $this->t('Styling'),
       '#description' => $this->t('Set to False to omit instagramFeed styles and provide your own in your CSS.'),
-      '#default_value' => isset($config['simple_instagram_styling']) ? $config['simple_instagram_styling'] : 'true',
+      '#default_value' => isset($config['simple_instagram_styling']) ? $config['simple_instagram_styling'] : 'false',
     ];
 
     return $form;
@@ -105,6 +104,7 @@ class SimpleInstagramBlock extends BlockBase implements BlockPluginInterface {
       '#cache' => [
         'max-age' => 3600,
       ],
+      'simple_instagram_username' => ['#markup' => $this->configuration['simple_instagram_username']],
     ];
   }
 
