diff --git a/core/modules/field_ui/src/Element/FieldUiTable.php b/core/modules/field_ui/src/Element/FieldUiTable.php
index 10200fdd38..2a33462a8d 100644
--- a/core/modules/field_ui/src/Element/FieldUiTable.php
+++ b/core/modules/field_ui/src/Element/FieldUiTable.php
@@ -68,39 +68,41 @@ public static function tablePreRender($elements) {
           $parents[$name] = $parent ? array_merge($parents[$parent], [$parent]) : [];
           unset($list[$name]);
 
-          // Determine the region for the row.
-          $region_name = call_user_func_array($row['#region_callback'], [&$row]);
-
-          // Add the element in the tree.
-          // phpcs:ignore DrupalPractice.CodeAnalysis.VariableAnalysis.UnusedVariable
-          $target = &$trees[$region_name][''];
-          foreach ($parents[$name] as $key) {
-            $target = &$target['children'][$key];
-          }
-          $target['children'][$name] = ['name' => $name, 'weight' => $row['weight']['#value']];
-
-          // Add tabledrag indentation to the first row cell.
-          if ($depth = count($parents[$name])) {
-            $children = Element::children($row);
-            $cell = current($children);
-            $indentation = [
-              '#theme' => 'indentation',
-              '#size' => $depth,
-              '#suffix' => $row[$cell]['#prefix'] ?? '',
-            ];
-            $row[$cell]['#prefix'] = \Drupal::service('renderer')->render($indentation);
-          }
+          if (!empty($row['#region_callback'])) {
+            // Determine the region for the row.
+            $region_name = call_user_func_array($row['#region_callback'], [&$row]);
+
+            // Add the element in the tree.
+            // phpcs:ignore DrupalPractice.CodeAnalysis.VariableAnalysis.UnusedVariable
+            $target = &$trees[$region_name][''];
+            foreach ($parents[$name] as $key) {
+              $target = &$target['children'][$key];
+            }
+            $target['children'][$name] = ['name' => $name, 'weight' => $row['weight']['#value']];
+
+            // Add tabledrag indentation to the first row cell.
+            if ($depth = count($parents[$name])) {
+              $children = Element::children($row);
+              $cell = current($children);
+              $indentation = [
+                '#theme' => 'indentation',
+                '#size' => $depth,
+                '#suffix' => $row[$cell]['#prefix'] ?? '',
+              ];
+              $row[$cell]['#prefix'] = \Drupal::service('renderer')->render($indentation);
+            }
 
-          // Add row id and associate JS settings.
-          $id = Html::getClass($name);
-          $row['#attributes']['id'] = $id;
-          if (isset($row['#js_settings'])) {
-            $row['#js_settings'] += [
-              'rowHandler' => $row['#row_type'],
-              'name' => $name,
-              'region' => $region_name,
-            ];
-            $js_settings[$id] = $row['#js_settings'];
+            // Add row id and associate JS settings.
+            $id = Html::getClass($name);
+            $row['#attributes']['id'] = $id;
+            if (isset($row['#js_settings'])) {
+              $row['#js_settings'] += [
+                'rowHandler' => $row['#row_type'],
+                'name' => $name,
+                'region' => $region_name,
+              ];
+              $js_settings[$id] = $row['#js_settings'];
+            }
           }
         }
       }
